examples/bigquery_user_info_updater_tool/bigquery_user_info_updater/tests/test_user_info_updater.py

```diff
@@ -0,0 +1,348 @@
+        temp_table_query_config.use_legacy_sql = False
+        temp_table_query = self.bq_client.query(
+            query='SELECT * FROM `{0:s}.{1:s}.{2:s}`'.format(
+                project_id,
+                self.dataset_id,
+                self.temp_user_info_updates_id
+            ),
+            job_config=temp_table_query_config,
+            location='US'
+        )
+        temp_table_query.result()
+        temp_table_results_df = temp_table_query.to_dataframe()\
+            .sort_values(by=['userId']).reset_index(drop=True)
+
+        # Gather expected results for comparison
+        abs_path = os.path.abspath(os.path.dirname(__file__))
+        expected_temp_data_file = os.path.join(
+            abs_path,
+            'test_data/temp_user_info_updates_expected_1.csv'
+        )
+        expected_temp_table_df = pd.read_csv(expected_temp_data_file)
+        # convert ingestTimestamp to datetime
+        expected_temp_table_df['ingestTimestamp'] = pd.to_datetime(
+            expected_temp_table_df['ingestTimestamp']
+        )
+
+        # Compare results
+        pd.testing.assert_frame_equal(
+            temp_table_results_df,
+            expected_temp_table_df
+        )
+
+        # Query the final table to test that the merge_updates() function worked
+        final_table_query_config = bigquery.QueryJobConfig()
+        final_table_query_config.use_legacy_sql = False
+        final_table_query = self.bq_client.query(
+            query='SELECT * FROM `{0:s}.{1:s}.{2:s}`'.format(
+                project_id,
+                self.dataset_id,
+                self.user_info_final_id
+            ),
+            job_config=final_table_query_config,
+            location='US'
+        )
+        final_table_query.result()
+        final_table_results_df = final_table_query.to_dataframe() \
+            .sort_values(by=['userId']).reset_index(drop=True)
+        # Gather expected results for comparison
+        expected_final_data_file = os.path.join(
+            abs_path,
+            'test_data/user_info_final_expected_1.csv'
+        )
+        expected_final_table_df = pd.read_csv(expected_final_data_file)
+        expected_final_table_df['ingestTimestamp'] = pd.to_datetime(
+            expected_final_table_df['ingestTimestamp']
+        )
+
+        # Compare results
+        pd.testing.assert_frame_equal(
+            final_table_results_df,
+            expected_final_table_df
+        )
+
+    def test_second_update(self, project_id):
+        """Tests UserInfoUpdater ability to run an update after at least one
+            has been run prior.
+
+         Args:
+             project_id(str): ID of the project that holds the test BQ tables.
+
+         Returns:
+             True if test passes, else False.
+                 """
+        # Load a second set of user updates to user_info_updates table to
+        # simulate a second iteration.
+        self.load_csv_to_bq(
```

Comment:
jaketf on Aug 27, 2019 Member
can we DRY up this "load this file to bigquery" pattern as a convenience function?

def load_csv_to_bq(self, filename, table):
        job_config = bigquery.LoadJobConfig()
        job_config.source_format = bigquery.SourceFormat.CSV
        job_config.skip_leading_rows = 1

        abs_path = os.path.abspath(os.path.dirname(__file__))
                with open(temp_data_file, 'rb') as file_obj:
            load_job = self.bq_client.load_table_from_file(
                file_obj=file_obj,
                destination=self.dataset_ref.table(
                    table),
                job_config=job_config
            )
            return load_job.result()